File: LayoutTests/http/tests/cache/resources/iframe304body.php
<?php
require_once '../../resources/portabilityLayer.php';

clearstatcache();

if ($_SERVER["HTTP_IF_MODIFIED_SINCE"]) {
    header("HTTP/1.1 304 Not Modified");
    exit();
}
header('Cache-Control: no-cache');
header('Content-Type: text/html');
header('Etag: 123456789');
?>
body text

File: LayoutTests/http/tests/cache/resources/iframe304body.py
#!/usr/bin/env python3

import os
import sys
sys.stdout.reconfigure(newline="")  # prevent windows \n -> \n\r conversion

modified_since = os.environ.get('HTTP_IF_MODIFIED_SINCE', '')

sys.stdout.write('Content-Type: text/html\r\n')

if modified_since:
    sys.stdout.write('status: 304\r\n\r\n')
    sys.exit(0)

sys.stdout.write(
    'Cache-Control: no-cache\r\n'
    'Etag: 123456789\r\n\r\n'
    'body text'
)
